diff --git a/3rdparty/dear-imgui/imgui_user.h b/3rdparty/dear-imgui/imgui_user.h
index e1819956f..1c62156a8 100644
--- a/3rdparty/dear-imgui/imgui_user.h
+++ b/3rdparty/dear-imgui/imgui_user.h
@@ -46,7 +46,7 @@ namespace ImGui
 #include "widgets/color_wheel.h"
 #include "widgets/dock.h"
 #include "widgets/file_list.h"
-#include "widgets/gizmo.h"
+// #include "widgets/gizmo.h"
 #include "widgets/markdown.h"
 #include "widgets/memory_editor.h"
 #include "widgets/range_slider.h"
diff --git a/3rdparty/dear-imgui/imgui_user.inl b/3rdparty/dear-imgui/imgui_user.inl
index 1ef190276..e25478bc8 100644
--- a/3rdparty/dear-imgui/imgui_user.inl
+++ b/3rdparty/dear-imgui/imgui_user.inl
@@ -75,7 +75,7 @@ namespace ImGui
 #include "widgets/color_wheel.inl"
 #include "widgets/dock.inl"
 #include "widgets/file_list.inl"
-#include "widgets/gizmo.inl"
+// #include "widgets/gizmo.inl"
 #include "widgets/markdown.inl"
 #include "widgets/memory_editor.inl"
 #include "widgets/range_slider.inl"
diff --git a/examples/common/imgui/imgui.cpp b/examples/common/imgui/imgui.cpp
index 4cf1c4f10..ea6c28856 100644
--- a/examples/common/imgui/imgui.cpp
+++ b/examples/common/imgui/imgui.cpp
@@ -408,7 +408,7 @@ struct OcornutImguiContext
 
 		ImGui::NewFrame();
 
-		ImGuizmo::BeginFrame();
+		// ImGuizmo::BeginFrame();
 	}
 
 	void endFrame()
