diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ff3401..4900392 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,8 +215,11 @@ if(BROTLI_PARENT_DIRECTORY)
 endif()
 
 # Build the brotli executable
-add_executable(brotli ${BROTLI_CLI_C})
-target_link_libraries(brotli ${BROTLI_LIBRARIES_STATIC})
+option(BROTLI_BUILD_TOOLS "Build executable brotli tool" ON)
+if(BROTLI_BUILD_TOOLS)
+  add_executable(brotli ${BROTLI_CLI_C})
+  target_link_libraries(brotli ${BROTLI_LIBRARIES_STATIC})
+endif()
 
 # Installation
 if(NOT BROTLI_EMSCRIPTEN)
