diff --git a/CMakeLists.txt b/CMakeLists.txt
index 4ff3401..14179fc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -215,16 +215,21 @@ if(BROTLI_PARENT_DIRECTORY)
 endif()
 
 # Build the brotli executable
-add_executable(brotli ${BROTLI_CLI_C})
-target_link_libraries(brotli ${BROTLI_LIBRARIES_STATIC})
+option(BROTLI_BUILD_TOOLS "Build executable brotli tool" ON)
+if(BROTLI_BUILD_TOOLS)
+  add_executable(brotli ${BROTLI_CLI_C})
+  target_link_libraries(brotli ${BROTLI_LIBRARIES_STATIC})
+endif()
 
 # Installation
 if(NOT BROTLI_EMSCRIPTEN)
 if(NOT BROTLI_BUNDLED_MODE)
+  if(BROTLI_BUILD_TOOLS)
   install(
     TARGETS brotli
     RUNTIME DESTINATION "${CMAKE_INSTALL_BINDIR}"
   )
+  endif()
 
   install(
     TARGETS ${BROTLI_LIBRARIES_CORE}
